specs/GetFavicon.md

```diff
@@ -0,0 +1,155 @@
+# Background
+A favicon (favorite icon) is a tiny icon included along with a website, which is displayed in places like the browser's address bar, page tabs and bookmarks menu. Developers would like to have an API which allows them to retrieve the Favicon of a webpage, if it has been set, as well as get an update whenever the favicon has changed.
+
+# Description
+We propose a new Webview2 event which would allow developers to access the current Favicon of a page and be notified when the favicon changes. This means when a page first loads, it would raise the FaviconChanged event since before parsing the HTML document there is no known favicon. DOM or JavaScript may change the Favicon, causing the event to be raised again for the same document.
+
+# Examples
+## Win32 C++ Registering a listener for favicon changes
+```cpp
+CHECK_FAILURE(m_webView->add_FaviconChanged(
+    Callback<ICoreWebView2FaviconChangedEventHandler>(
+    [this](ICoreWebView2* sender, IUnknown* args) -> HRESULT {
+
+    wil::com_ptr<IStream> iconStream = SHCreateMemStream(nullptr, 0);
+
+    sender->GetFavicon(COREWEBVIEW2_FAVICON_IMAGE_FORMAT_PNG, iconStream.get(),
+        Callback<ICoreWebView2ExperimentalGetFaviconCompletedHandler>(
+            [iconStream, this](HRESULT error_code) -> HRESULT
+            {
+                if (error_code == S_OK)
+                {
+                    Gdiplus::Bitmap* iconBitmap =
+                        new Gdiplus::Bitmap(iconStream.get());
+                    HICON icon;
```

Comment:
This code leaks the icon. Note that the icon handle needs to remain valid for as long as it is the window's current icon, so it needs to be held in a member variable.

```cpp
/* member variable */
wil::unique_hicon m_favicon;

/* completion handler */
Gdiplus::Bitmap iconBitmap(iconStream.get());
wil::unique_hicon icon;
if (iconBitmap.GetHICON(&icon)) {
    m_favicon = std::move(icon);
    SendMessage(m_appWindow->GetMainWindow(), WM_SETICON, ICON_SMALL, (LPARAM)m_favicon.get());
}
}
```

Reply:
Thanks for the catch on this.